### PR TITLE
keyutils-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 396b6a9249642f05dd02ef9e5c1a59788bcbbfd1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T05.20-396b6a9/job.log
 (1/2) keyutils-tests.py:KeyUtils.test;run-run_type-upstream-ddda:  FAIL: 86 test(s) failed, please refer to the log (140.40 s)
 (2/2) keyutils-tests.py:KeyUtils.test;run-run_type-distro-f9f8:  FAIL: keyutils-tests.py: 'make check' failed. (6.41 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T05.20-396b6a9/results.html
JOB TIME   : 158.01 s


SLES:
JOB ID     : d0d295134be9f5405fa766050e0d8a939d23b14f
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T15.48-d0d2951/job.log
 (1/2) keyutils-tests.py:KeyUtils.test;run-run_type-upstream-ddda:  FAIL: 5 test(s) failed, please refer to the log (153.40 s)
 (2/2) keyutils-tests.py:KeyUtils.test;run-run_type-distro-f9f8:  FAIL: keyutils-tests.py: 'make check' failed. (9.38 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T15.48-d0d2951/results.html
JOB TIME   : 181.89 s
```

File: security/keyutils-tests.py [hunk 1]
```diff
@@ -35,19 +35,33 @@ def setUp(self):
...
-        os.environ['AUTOMATED'] = '1'
-        self.make_path = os.path.join(self.workdir, 'tests')
-        os.chdir(self.make_path)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = ("https://git.kernel.org/pub/scm/linux/kernel/git/"
+                       "dhowells/keyutils.git")
+            url = self.params.get('url', default=def_url)
+            git.get_repo(url, destination_dir=self.workdir)
+            os.environ['AUTOMATED'] = '1'
```
Comment (Naresh-ibm) [inline security/keyutils-tests.py:44]: @nasastry is this not required for distro type?

Comment (nasastry) [reply security/keyutils-tests.py:44]: @Naresh-ibm nice catch, by make the above line common for 'upstream' and 'distro' ... distro test cases are executing properly.
Thanks!! Sending the updated patch.

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : c385d9501076bcc7b4c99ef19be4ce27246725ef
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T04.56-c385d95/job.log
 (1/2) keyutils-tests.py:KeyUtils.test;run-run_type-upstream-ddda: FAIL: 86 test(s) failed, please refer to the log (141.16 s)
 (2/2) keyutils-tests.py:KeyUtils.test;run-run_type-distro-f9f8: PASS (7.38 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T04.56-c385d95/results.html
JOB TIME   : 159.69 s


SLES:
JOB ID     : 54ec0967ad5478c54d5d2e2107be7514a1c46ef9
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-28T15.24-54ec096/job.log
 (1/2) keyutils-tests.py:KeyUtils.test;run-run_type-upstream-ddda: FAIL: 5 test(s) failed, please refer to the log (153.92 s)
 (2/2) keyutils-tests.py:KeyUtils.test;run-run_type-distro-f9f8: FAIL: 5 test(s) failed, please refer to the log (160.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-28T15.24-54ec096/results.html
JOB TIME   : 333.05 s
```

File: security/keyutils-tests.py [hunk 2]
```diff
@@ -35,19 +35,33 @@ def setUp(self):
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git"
-        git.get_repo(url, destination_dir=self.workdir)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = ("https://git.kernel.org/pub/scm/linux/kernel/git/"
```
Comment (abdhaleegit) [inline security/keyutils-tests.py:40]: When yaml has hardcoded url, than you can remove the url here

File: security/keyutils-tests.py [hunk 3]
```diff
@@ -35,19 +35,33 @@ def setUp(self):
...
     def test(self):
         '''
         Running tests from keyutils
         '''
         count = 0
-        output = build.run_make(self.make_path,
+        output = build.run_make(self.srcdir,
                                 process_kwargs={"ignore_status": True})
+        if output.exit_status:
+            self.fail("keyutils-tests.py: 'make check' failed.")
```
Comment (abdhaleegit) [inline security/keyutils-tests.py:64]: If we remove ignore status than the above exit status check can be removed

File: security/keyutils-tests.py.data/keyutils.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+run_type: !mux
+    upstream:
+        type: 'upstream'
+    distro:
+        type: 'distro'
+url: "https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git"
```
Comment (abdhaleegit) [inline security/keyutils-tests.py.data/keyutils.yaml:6]: Ok with the above yaml, bothe distro and upstream cases gets executed every time

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: security/keyutils-tests.py.data/keyutils.yaml (hunk 4 above)
Comment (abdhaleegit) [reply security/keyutils-tests.py.data/keyutils.yaml:6]: Looks good, but the other way would be of url is given assume as upstream run else go for distro run   .. So in that case the type param can be eliminated

Comment (Naresh-ibm) [reply security/keyutils-tests.py.data/keyutils.yaml:6]: from code, url is required for `upstream` type only.
so we can remove the default url and hard_code it in yaml.
@nasastry

**Review by abdhaleegit — APPROVED** [review]
@Naresh-ibm as discussed offline with @nasastry basically the idea to hardcode url in both yaml and also py file.. is to make sure the test runs even with out yaml file

**Review by Naresh-ibm — APPROVED** [review]
LGTM
@abdhaleegit Thanks for Review and @nasastry thanks for PR

**Pull request merged**